stan/math/prim/mat/fun/cov_exp_quad.hpp

```diff
@@ -66,75 +74,86 @@ inline
 }
 
 /**
- * Returns a squared exponential kernel.
+ * Returns a squared exponential kernel, with automatic relevance determination,
+ * (ARD), a seperate length scale for each dimension.
+ *
+ *  \f$k(x,x') = \sigma^2 exp( \frac{d(x, x')^2}{2l_d^2})  \f$
+ *
+ *  where \f$d(x, x)\f$ is euclidean distance, and \f$d\f$ is a subscript for
+ *  each dimension.
  *
  * @tparam T_x type of std::vector of elements
  * @tparam T_sigma type of sigma
  * @tparam T_l type of std::vector of length scale
  *
  * @param x std::vector of elements that can be used in square distance.
  *    This function assumes each element of x is the same size.
- * @param sigma standard deviation
+ * @param sigma marginal standard deviation, or magnitude.
  * @param length_scale std::vector length scale
  * @return squared distance
- * @throw std::domain_error if sigma <= 0, l <= 0, or
+ * @throw std::domain_error if sigma <= 0, length_scale <= 0, or
  *   x is nan or infinite
+ * @throw std::invald_arguemnt if dimension of x is not the same
+ *   length as length_scale
  */
 template <typename T_x, typename T_sigma, typename T_l>
-inline
-    typename Eigen::Matrix<typename stan::return_type<T_x, T_sigma, T_l>::type,
-                           Eigen::Dynamic, Eigen::Dynamic>
-    cov_exp_quad(const std::vector<T_x>& x, const T_sigma& sigma,
-                 const std::vector<T_l>& length_scale) {
+inline typename boost::enable_if_c<
```

Comment:
Why is this `enable_if`'d?

Reply:
If this is limiting `T_x`, then please update the doc appropriately for `@tparam T_x`.

Reply:
I enabled if'd it because having seperate length scales for each dimension is only possible when D > 1, which is only possible if T_x in this case is a vector like. Is this necessary? I think then the function will only be called if length scale is a vector, so the function call is unique, so I think it's fine. I'm just removing all the enable ifs